Services/GEV/Desktop/classes/EffectivenessAnalysis/class.gevMyEffectivenessAnalysisGUI.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/* Copyright (c) 1998-2016 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+* @ilCtrl_Calls gevMyEffectivenessAnalysisGUI: gevEffectivenessAnalysisGUI
+*/
+class gevMyEffectivenessAnalysisGUI {
+	const CMD_VIEW = "view";
+	const CMD_FILTER = "filter";
+	const CMD_SEARCH_USER = "userfieldAutocomplete";
+
+	/**
+	 * @var ilCrl
+	 */
+	protected $gCtrl;
+
+	/**
+	 * @var ilTemplate
+	 */
+	protected $gTpl;
+
+	/**
+	 * @var ilLanguage
+	 */
+	protected $gLng;
+
+	public function __construct() {
+		global $ilCtrl, $tpl, $lng, $ilUser;
+
+		$this->gCtrl = $ilCtrl;
+		$this->gTpl = $tpl;
+		$this->gLng = $lng;
+		$this->gUser = $ilUser;
+
+		require_once("Services/GEV/Desktop/classes/EffectivenessAnalysis/class.gevEffectivenessAnalysis.php");
+		$this->eff_analysis = gevEffectivenessAnalysis::getInstance();
+
+		$this->eff_analysis_icon = '<img src="'.ilUtil::getImagePath("GEV_img/ico_eff_analysis.png").'" />';
+	}
+
+	public function executeCommand() {
+		$cmd = $this->gCtrl->getCmd(self::CMD_VIEW);
+		$next_class = $this->gCtrl->getNextClass();
+
+		switch($next_class) {
+			case "geveffectivenessanalysisgui":
+				require_once("Services/GEV/Desktop/classes/EffectivenessAnalysis/class.gevEffectivenessAnalysisGUI.php");
+				$gui = new gevEffectivenessAnalysisGUI();
+				$this->gCtrl->forwardCommand($gui);
+				break;
+			default:
+				switch($cmd) {
+					case self::CMD_VIEW:
+					case self::CMD_FILTER:
+						$this->view();
+						break;
+					case self::CMD_SEARCH_USER:
+						$this->$cmd();
+						break;
+				default:
+			}
+		}
+	}
+
+	protected function view() {
+		$html = $this->getTitle()->render();
+
+		$filter = $this->getFilter();
+		$filter_values = $this->eff_analysis->buildFilterValuesFromFilter($filter);
+		$html .= $filter->render();
+
+		$this->gCtrl->setParameter($this, $filter->getGETName(), $filter->encodeSearchParamsForGET());
+		$html .= $this->getTable($filter_values)->getHtml();
+		$this->gCtrl->setParameter($this, $filter->getGETName(), null);
+
+		$this->gTpl->setContent($html);
+	}
+
+	protected function getFilter() {
+		$filter = $this->eff_analysis->getFilter($this->gUser->getId());
```

Comment:
Hier könntest du natrülich auch direkt return $this->eff_analysis->getFilter($this->gUser->getId())->action ...

Aber ist Geschmacksache